pkg/rsstorage/servers/s3server/s3_enumeration.go

```diff
@@ -176,3 +285,21 @@ func getObjectsAll(bucketObjectsList *s3.ListObjectsOutput, s3Prefix string, reg
 
 	return binaryMeta
 }
+
+func getObjectsAllMap(bucketObjectsList *s3.ListObjectsOutput, s3Prefix string, reg *regexp.Regexp) map[string]string {
+	binaryMeta := make(map[string]string)
+
+	for _, key := range bucketObjectsList.Contents {
+
+		if reg != nil {
+			if s := reg.FindStringSubmatch(*key.Key); len(s) > 1 {
+				binaryMeta[strings.TrimPrefix(s[1], s3Prefix)] = *key.ETag
+			}
+		} else {
+			binaryMeta[strings.TrimPrefix(*key.Key, s3Prefix)] = *key.ETag
```

Comment:
This method is nearly identical to the existing `getObjectsAll` method except that it returns a map and sets the map key/value pairs here instead of simply appending to a string array.